inteligent_scissors/fLiveWireGetCostFcn.m

```diff
@@ -0,0 +1,21 @@
+function dF = fLiveWireGetCostFcn(dImg, dWz, dWg, dWd)
+
+if nargin < 2,
+    dWz = 0.2;
+    dWg = 0.8;
+    dWd = 0.2;
+end
+
+% Calculat the cost function
+% The gradient strength cost Fg
+dImg = double(dImg);
+[dY, dX] = gradient(dImg);
+dFg = sqrt(dX.^2 + dY.^2);
+dFg = 1 - dFg./max(dFg(:));
+
+% The zero-crossing cost Fz
+lFz = ~edge(dImg, 'zerocross');
+
+% The gradient direction Fd ??
+% The Sum:
+dF = dWz.*double(lFz)+ dWg.*dFg;

```

Comment:
Do not forget that costs are defined at pixel p from its 8 neighbors (horizontal, vertical and diagonal). Fg is for instance scaled by 1 for diagonal neighbors and by 1/sqrt(2) for horizontal and vertical neighbors. See the paper (eq. 3 and text around).